litmus-portal/backend/self-deployer/pkg/util/uitls.go

```diff
@@ -0,0 +1,101 @@
+package util
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/litmuschaos/litmus/litmus-portal/backend/self-deployer/pkg/k8s"
+
+)
+
+func WaitForServer(server string) {
+	log.Print("Checking if Server Up ", server)
+	for true {
```

Comment:
Will this going to be an infinite loop if the server responds with 200? Maybe you can use a timeout instead of a loop.
Say if the server didn't come up than the self-deployer pod will go `crashbackloop `state, so the user will get to know about the problem. Otherwise, the pod will be in running state.

Reply:
Currently what I am doing is that the self-deployer will loop and check if the GQL server responds 200 or not. If it doesn't I sleep for 30sec and then try again. It is true that it will keep doing this forever if the server doesn't come up, but most probably the server will come up and it will exit the loop, or else the user will notice that there's a error with the portal and debug the deployments. I can try applying some timeout too, but what would be a good timeout period? Like it may happen the server takes some time to come up depending on the resources available.

Reply:
Having a timeout will be good to make the pod crashbackloop to inform  the user about the problem.

Reply:
Max timeout can be 5 mins

Reply:
Okay I'll try adding that, I'll set a panic when the timeout happens. that should give a crashbackloop